### DISCUSSION TITLE
hwtest fail with 0% bad

### DISCUSSION THREAD

**Discussion by brucke** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#579] [Q&A]
Hello Community, 

after some hours of reading the documentation and and soldering I'm abel to run the hwtest.
But i have one issue. 

The connection between CIO to DSP pin 5 was not succesful, 

> 0% bad

I have no idea what the problem is, i check all soldering point especially related to pin 5. 

Before connectin to the real pump, this test should be succesful, 
Does anybody got an idea / hint for me?
The main question is, is there a problem with the soldering points? And which shall i check?

Thanks in advance. 
Bernhard

![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6576396/d76d2d0d-42c8-42ef-b8dd-b25e3f1b79a7)

![PXL_20230728_140113767](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6576396/3cee43da-5d07-48df-a36a-fff35f9b63eb)

![PXL_20230728_140103452](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6576396/a584ad79-9179-468c-95d6-0c9346c7299a)

![PXL_20230728_140059731](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6576396/1dd99a88-4ebc-49ad-bb92-f04cf084e664)

**Comment by visualapproach** [comment]
Hi! I'm happy to see you used headers for the esp and LLC. It makes it much easier to swap the parts to conclude what's wrong. You have a peculiar error. The test shows that the connection works one way but not the other. If it's consistent the output on cio side or input on dsp side may be faulty. So if you have spare parts the easiest way to test is to swap the esp to another one, and LLC too, to know what is wrong.

**Comment by brucke** [comment]
Good Morning @visualapproach , thanks for your answer! 😄

it did following test: 

Change Pinout from 


**CIO**
| name       | V2B | Custom |
| ---------- | --- | ------ |
| data/td/rx | 1   | 1      |
| clk/tx     | 2   | 5      |
| cs/ld      | 5   | 2      |

**DSP**
| name       | V2B | Custom |
| ---------- | --- | ------ |
| data/td/rx | 6   | 6      |
| clk/rx     | 4   | 3      |
| cs/ld      | 3   | 4      |

And now i got following error

![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6576396/7e2e8707-e0bd-4223-b79b-637ccca318d2)

My assumption is the ESP is broken. If i change the output pin on ESP the failed pin changes also. 
In case of broken TXS the PIN would always be the same. 

Is my assumption right? Or did i miss something?

thanks
Bernhard

**Reply by visualapproach** [reply]
I'd try another LLC then ESP. It can still be the LLC because D5/2 is not moving by swapping them in the UI. You just lie to the ESP about where each wire goes.

**Reply by brucke** [reply]
I ordered a new LLC. 
i will do the next try on monday :) 
thanks, have a nice weekend.

**Comment by brucke** [comment]
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6576396/ba01ea64-3b26-4342-99d4-7eb527b388af)
Thanks now it's working!!
